update_bundle.py

```diff
@@ -3,42 +3,57 @@
 # See LICENSE file for licensing details.
 
 import logging
-import requests
-import yaml
+import subprocess
 import sys
-
 from pathlib import Path
 
+import requests
+import yaml
+
 logger = logging.getLogger(__name__)
 
 
-def fetch_revision(charm, charm_channel):
+def fetch_revision(charm, charm_channel, ubuntu_version=None):
     """Returns revision number for charm in channel."""
     charm_info = requests.get(
         f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map"
     ).json()
 
     [track, risk] = charm_channel.split("/")
 
+    revisions = []
     for channel in charm_info["channel-map"]:
         if channel["channel"]["risk"] == risk and channel["channel"]["track"] == track:
-            return channel["revision"]["revision"]
+            if ubuntu_version is None or ubuntu_version == channel["channel"]["base"]["channel"]:
+                revisions.append(channel["revision"]["revision"])
+    if revisions:
+        # If the charm supports multiple Ubuntu bases, it's possible
+        # that there is a different revision for each base.
+        # Select the latest revision.
+        return max(revisions)
+
+    raise ValueError(f"Revision not found for {charm} on {charm_channel} for Ubuntu {ubuntu_version}")
 
-    raise ValueError(f"Revision not found for {charm} on {charm_channel}")
 
 def update_bundle(bundle_path):
     """Updates a bundle's revision number."""
     bundle_data = yaml.safe_load(Path(bundle_path).read_text())
 
     for applications in bundle_data["applications"]:
+        if series := bundle_data["applications"][applications].get("series"):
+            ubuntu_version = subprocess.check_output(
+                ["ubuntu-distro-info", "--series", series, "--release"], encoding="utf-8"
```

Comment:
thoughts on passing in the series as an argument to this script instead of inferring it from the distro of the GH runner? i can see it being confusing if we upgrade the charms' series but do not know to update the GH runner version? or if the `ubuntu-latest` gets updated but we have not yet updated the charms' series

Reply:
sorry, i was mistaken about what `ubuntu-distro-info` does. the change makes more sense now, thanks!

Reply:
Yes, cli argument was a good option too.. but merged already.
Let's see how stable the current implementation is and refactor to cli if necessary.
Fully automated approach is always better if it works reliable. :-D